sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc

```diff
@@ -232,46 +232,17 @@ class annotated_arg {
     template<typename propertyT>
     static constexpr /*unspecified*/ get_property();
 
-    // Overloaded arithmetic operators
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() + std::declval<T2>())>
-    R operator+(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() - std::declval<T2>())>
-    R operator-(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() * std::declval<T2>())>
-    R operator*(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() / std::declval<T2>())>
-    R operator/(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() % std::declval<T2>())>
-    R operator%(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() & std::declval<T2>())>
-    R operator&(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() | std::declval<T2>())>
-    R operator|(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() ^ std::declval<T2>())>
-    R operator^(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() >> std::declval<T2>())>
-    R operator>>(const annotated_arg<T2, PropertyList2> &other) const;
-
-    template <typename T2, typename PropertyList2,
-              typename R = decltype(std::declval<T>() << std::declval<T2>())>
-    R operator<<(const annotated_arg<T2, PropertyList2> &other) const;
+    // OP is: +, -, *, /, %, <<, >>, &, |, ^, <, <=, >, >=, ==, ~=, &&, ||
```

Comment:
Please remind me: Why do we do operator forwarding for annotated_arg? For annoted_ref it's needed because it doesn't have an implicit conversion. But this does. Why do we need operator forwarding nonetheless?

Reply:
This https://godbolt.org/z/h5cTTr17K would be a good example to show why this is not working without this fix.

Reply:
Thanks. Makes sense.